### ISSUE TITLE
Is it possible to get the category of a predicted box when calling MMOCR()

### ISSUE THREAD

**Issue by CodePothunter** [open-mmlab/mmocr#1050]
I am looking for a way to show the predicted category of the output from the textdet mode. Currently, the parameter `details=True` only produces `boundary_result`. 

**Is it possible to get the category of a predicted box when calling MMOCR()?**

**Comment by Mountchicken** [comment]
Hi @CodePothunter 
I'm a little confused by your question. There is only one output category for text detection, and that is text.

**Comment by CodePothunter** [comment]
> Hi @CodePothunter I'm a little confused by your question. There is only one output category for text detection, and that is text.

Thank you for your quick reply. I re-trained the det. model (i.e., FCENet) with my customized dataset of multiple categories. As I observed in the training log, there were loss terms regarding classification accuracy. Therefore, I thought that at least in mmdetection module, it is possible to achieve the predicted classes.

**Comment by Mountchicken** [comment]
It feels like this will be very complicated. Do you have any idea ?@gaotongxiao

**Comment by CodePothunter** [comment]
> I don't think we have any loss function concerning classification accuracy. To maintain some compatibility with MMDetection, MMOCR defines only one category "text", which however does not provide any information to text detection models as these models only focus on telling the text region apart from the background. Therefore, text detection models don't really have a classification loss as in MMDetection, except for MaskRCNN, which actually was adapted from MMDetection.
> 
> > As I observed in the training log, there were loss terms regarding classification accuracy.
> 
> I guess what you've seen were the recalls and precisions reported by our hmean-iou metric. These statistics come from the overlaps between predicted & ground truth polygons instead of object categories.

I see. I came up with the idea of training each category per mmdet model, which might provide flexible usage under the current design of MMOCR. Would this be a good practice?

**Comment by Mountchicken** [comment]
You can refer to [issue1027](https://github.com/open-mmlab/mmocr/issues/1027), which also has to do some pre-processing on the detection results before sending them to a recognizer.

**Comment by CodePothunter** [comment]
> You can refer to [issue1027](https://github.com/open-mmlab/mmocr/issues/1027), which also has to do some pre-processing on the detection results before sending them to a recognizer.

Thank you for your reply. That's what I need.
